src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/hmi/vr_add_command_request.cc

```diff
@@ -57,7 +57,19 @@ VRAddCommandRequest::~VRAddCommandRequest() {}
 void VRAddCommandRequest::Run() {
   SDL_LOG_AUTO_TRACE();
 
-  SendRequest();
+  if ((*message_)[strings::msg_params].keyExists(strings::vr_commands)) {
+    SendRequest();
+  } else {
+    event_engine::Event event(hmi_apis::FunctionID::VR_AddCommand);
+
+    (*message_)[strings::params][strings::message_type] =
+        static_cast<int32_t>(application_manager::MessageType::kResponse);
+    (*message_)[strings::params][application_manager::hmi_response::code] =
+        static_cast<int32_t>(hmi_apis::Common_Result::SUCCESS);
+
+    event.set_smart_object(*message_);
+    event.raise(application_manager_.event_dispatcher());
+  }
```

Comment:
I would remove all of these other changes from the PR, we shouldn't encounter these cases with my suggested changes